### DISCUSSION TITLE
A prefix sum in ISPC

### DISCUSSION THREAD

**Discussion by jacobloveless** [ispc/ispc#2436] [Q&A]
I realize this is a bit open - but can anyone express an SIMD prefix sum (aka cumulative sum) in ISPC?

ref: https://en.algorithmica.org/hpc/algorithms/prefix/

**Comment by jacobloveless** [comment]
If anyone has an example of a (correctly vectorized) version of the prefix sum - that would be immensely helpful!
We are finding that expressing this in ISPC is extraordinarily difficult. Thanks everyone!

**Comment by JebScape** [comment]
exclusive_scan_add() already exists within the standard library that does this for you.
![IMG_8225](https://github.com/ispc/ispc/assets/112780479/26392263-d237-4d7b-98d3-c4b87aaaa4f9)
